src/scala/microsoft-spark-2.4.x/src/main/scala/org/apache/spark/api/dotnet/DotnetBackendHandler.scala

```diff
@@ -222,7 +222,7 @@ class DotnetBackendHandler(server: DotnetBackend)
       return false
     }
 
-    for (i <- 0 to numArgs - 1) {
+    for (i <- 0 until numArgs) {
```

Comment:
Since `until` has been changed here in `2.4.x`, we should also modify `2.3.x` and `3.0.x` to keep consistent,

Reply:
Done, thanks!